scripts.py

```diff
@@ -292,12 +302,59 @@ def __init__(self, provider_class, _db=None, providers=None, **kwargs):
         providers = providers or list()
         if provider_class:
             providers += self.get_providers(_db, provider_class, **kwargs)
-        super(RunCollectionCoverageProviderScript, self).__init__(providers)
+        super(RunCollectionCoverageProviderScript, self).__init__(providers, _db=_db)
 
     def get_providers(self, _db, provider_class, **kwargs):
         return list(provider_class.all(_db, **kwargs))
 
 
+class RunThreadedCollectionCoverageProviderScript(Script):
+    """Run coverage providers in multiple threads."""
+
+    DEFAULT_WORKER_SIZE = 5
+
+    def __init__(self, provider_class, worker_size=None, _db=None, **provider_kwargs):
+        super(RunThreadedCollectionCoverageProviderScript, self).__init__(_db)
+
+        self.session_factory = SessionManager.sessionmaker(session=self._db)
+        self.worker_size = worker_size or self.DEFAULT_WORKER_SIZE
+
+        self.provider_class = provider_class
+        self.provider_kwargs = provider_kwargs
+
+    def worker_factory(self, job_queue):
+        """Returns a Worker with a database"""
+        return DatabaseWorker(job_queue, scoped_session(self.session_factory))
+
+    def run(self):
+        _db = scoped_session(self.session_factory)
+        collections = self.provider_class.collections(_db)
+
+        if not collections:
+            return
+
+        with Pool(self.worker_size, self.worker_factory) as pool:
+            for collection in collections:
+                offset = 0
+                query_size, batch_size = self.get_query_and_batch_sizes(
+                    collection
+                )
+                _db.commit()
```

Comment:
`get_query_and_batch_sizes` doesn't seem like a method name that needs a commit. Presumably the commit is because it creates a provider object to estimate the collection size, and that  might do anything. Could you add a comment to this effect?